### ISSUE TITLE
[Bug] Bad file name when downloading a file from a blob: url

### ISSUE THREAD

**Issue by bonninf** [duckduckgo/Android#2598]
### Describe the bug

When a file is downloaded from a blob: url then the file is saved into the device with a wrong file name. With DDG the name is a guid such as: "b394...0b76.pdf" whereas the name is such as "attestation-2022-12-04_11-16.pdf" with another browser as Chrome.

Furthermore the time to download the file is very long (a couple of seconds) with DDG whereas it's almost instantaneous by using another browser as Chrome for instance.

### How to Reproduce

1. Go to the url: https://media.interieur.gouv.fr/attestation-deplacement-derogatoire-covid-19/ 
2. Fill in the form with valid data (see screenshot)
3. Touch the button to generate the attestation: "Générer mon attestation" (see screenshot)
4. The Web page shows the document is downloaded: "L'attestation is téléchargée sur votre appareil" (see screenshot)
5. However the document is not yet available on the device and the DDG's Downloads section does not contain the document yet
6. We have to wait a couple of seconds yet to see the DDG's confirmation that the file is downloaded
7. At this time the downloaded document is well available from the DDG's Downloads section (see screenshot) but the name is wrong

![image](https://user-images.githubusercontent.com/1090995/205488584-0808af15-89e0-4570-80c1-669cd7997f4a.png)
![image](https://user-images.githubusercontent.com/1090995/205488609-9e157fd4-0b60-4f4f-8228-542662e6aaf9.png)
![image](https://user-images.githubusercontent.com/1090995/205488631-6b8d6e17-2627-4643-8671-0969c4533e96.png)
![image](https://user-images.githubusercontent.com/1090995/205488711-b3e8d78d-36ab-490e-9f4c-215602a87982.png)

### Expected behavior

The name of the saved file should be something as: "attestation-2022-12-04_11-16.pdf" (see the screenshot from Chrome below)

The time to download the file should be almost instantaneous.

![image](https://user-images.githubusercontent.com/1090995/205488536-80e61a4f-bc34-4aa3-9bae-55d461b1d5ef.png)

### Environment

```markdown
- DDG App Version: 5.144.0
- Device: Galaxy S10+ / Samsung SM-G975F/DS
- OS: Android 11
```

**Comment by cmonfortep** [comment]
Have you found a similar scenario on another site? That website doesn't seem something we can use for testing (as it requires valid info to download a covid certificate).

**Comment by anikiki** [comment]
@bonninf Thank you for raising this and posting all steps to reproduce the issue. After investigating it, we noticed that the name of the file comes as the value of the "download" attribute.

_<a href="blob:https://media.interieur.gouv.fr/f45f72d4-581a-4dcf-892b-511626ee63d5" download="attestation-2022-12-21_10-34.pdf"></a>_

Unfortunately, this is not made available by the Android webview and we can't access it. In this case we generate a random filename. Chrome and other browsers don't use the webview and have access to other means to get the correct file name.

Regarding the performance, we opened a PR https://github.com/duckduckgo/Android/pull/2673.
